### PR TITLE
Fix: linear accrual not overpassing maturity

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

[slack thread](https://kflabs.slack.com/archives/C04GJQAM9P0/p1715955299145219?thread_ts=1715935675.464829&cid=C04GJQAM9P0)

We need to avoid computing points in the line of linear accrual if the time overpass the maturity time, instead, we want to clamp the value to `notional` (which is the same when the time is just maturity)

### CODE REVIEW

File: pallets/loans/src/entities/loans.rs [hunk 1]
```diff
@@ -359,7 +359,8 @@ impl<T: Config> ActiveLoan<T> {
 				inner.adjust(Adjustment::Increase(amount.balance()?))?
 			}
 			ActivePricing::External(inner) => {
-				inner.adjust(Adjustment::Increase(amount.external()?), Zero::zero())?
+				let when = T::Time::now();
```
Comment (mustermeiszer) [inline pallets/loans/src/entities/loans.rs:362]: I guess maturity not being passed already is checked by the `ensure_can_borrw()`?

File: pallets/loans/src/tests/repay_loan.rs [hunk 2]
```diff
@@ -812,3 +812,41 @@ fn with_unregister_price_id_and_oracle_not_required() {
...
+
+		let active_loan = util::get_loan(loan_id);
+
+		let settlement_price_updated = match active_loan.pricing() {
+			ActivePricing::External(inner) => inner.settlement_price_updated(),
+			_ => unreachable!(),
+		};
+
+		// We must never overpass madurity date
+		assert_eq!(active_loan.maturity_date(), settlement_price_updated);
```
Comment (mustermeiszer) [inline pallets/loans/src/tests/repay_loan.rs:850]: Could we also check, that the linear pricing actually clamps with giving the notional baack?

File: pallets/loans/src/entities/loans.rs (hunk 1 above)
Comment (lemunozm) [reply pallets/loans/src/entities/loans.rs:362]: But not in `repay()`

**Comment by lemunozm** [comment]
BTW, I did the fix quite bad 😆. I'm redoing the fix implementation

File: pallets/loans/src/tests/repay_loan.rs (hunk 2 above)
Comment (lemunozm) [reply pallets/loans/src/tests/repay_loan.rs]: Done

File: pallets/loans/src/tests/repay_loan.rs [hunk 3]
```diff
@@ -812,3 +812,39 @@ fn with_unregister_price_id_and_oracle_not_required() {
...
+
+		assert_eq!(
+			ActiveLoanInfo::try_from((POOL_A, util::get_loan(loan_id)))
+				.unwrap()
+				.current_price
+				.unwrap(),
+			NOTIONAL
+		);
+	});
+}
```
Comment (mustermeiszer) [inline pallets/loans/src/tests/repay_loan.rs:850]: I would still like to test the case where 

`last_updated_price` and `T::Time::now()` are both equal to `maturity`. I think that is an edge case worth testing

**Comment by lemunozm** [comment]
Your migration should work without any change after rebasing @wischli 🤞🏻

File: pallets/loans/src/tests/repay_loan.rs (hunk 3 above)
Comment (lemunozm) [reply pallets/loans/src/tests/repay_loan.rs:850]: Well, it is repaid with one DAY of overdue, and the expected value is `notional`
